### ISSUE TITLE
Npcap Loopback adapter

### ISSUE THREAD

**Issue by RichardFoss** [sebhildebrandt/systeminformation#425]
Thanks for this great capability!

I want to discover and use the Ethernet wired interface for communication with audio devices.

I see that the Npcap Loopback adapter is discovered as a wired network interface. What is the best way to distinguish it from a 'hard' wired interface? I assumed that it might be viewed as 'internal', but this is not the case.

**Comment by sebhildebrandt** [comment]
@RichardFoss can you provide a little more information like which OS? I would be able to also ask for specific outputs. Would make it easier for me to provide a solution ;-)

**Comment by RichardFoss** [comment]
Thanks for the response Sebastian!
I am using Windows 10. This is the Npcap that, for example WireShark uses. 
Windows views it as the Npcap Loopback Adapter.
Currently, I am getting around this by discarding an interface that has the 'Loopback' string in its name. However, this is a bit of a hack, it would be preferable to pick up that this is a different type of adapter. What outputs would help you?
Thanks for your assistance!

**Comment by sebhildebrandt** [comment]
Thank you @RichardFoss , can you run the following code snippets on your machine:

in Nodejs (CLI)
```
os.networkInterfaces();
```

and in command line:

```
wmic nic get /value
```

**Comment by RichardFoss** [comment]
For nodejs cli:
> os.networkInterfaces()
{
  'Npcap Loopback Adapter': [
    {
      address: 'fe80::4cff:fe4f:4f50',
      netmask: 'ffff:ffff:ffff:ffff::',
      family: 'IPv6',
      mac: '02:00:4c:4f:4f:50',
      internal: false,
      cidr: 'fe80::4cff:fe4f:4f50/64',
      scopeid: 13
    },
    {
      address: '169.254.1.173',
      netmask: '255.255.0.0',
      family: 'IPv4',
      mac: '02:00:4c:4f:4f:50',
      internal: false,
      cidr: '169.254.1.173/16'
    }
  ],
  'Wi-Fi': [
    {
      address: '192.168.0.168',
      netmask: '255.255.255.0',
      family: 'IPv4',
      mac: '50:3e:aa:ef:ec:30',
      internal: false,
      cidr: '192.168.0.168/24'
    }
  ],
  'Loopback Pseudo-Interface 1': [
    {
      address: '::1',
      netmask: 'ffff:ffff:ffff:ffff:ffff:ffff:ffff:ffff',
      family: 'IPv6',
      mac: '00:00:00:00:00:00',
      internal: true,
      cidr: '::1/128',
      scopeid: 0
    },
    {
      address: '127.0.0.1',
      netmask: '255.0.0.0',
      family: 'IPv4',
      mac: '00:00:00:00:00:00',
      internal: true,
      cidr: '127.0.0.1/8'

For cmd line:
AdapterType=
AdapterTypeId=
AutoSense=
Availability=3
Caption=[00000000] Microsoft Kernel Debug Network Adapter
ConfigManagerErrorCode=0
ConfigManagerUserConfig=FALSE
CreationClassName=Win32_NetworkAdapter
Description=Microsoft Kernel Debug Network Adapter
DeviceID=0
ErrorCleared=
ErrorDescription=
GUID=
Index=0
InstallDate=
Installed=TRUE
InterfaceIndex=8
LastErrorCode=
MACAddress=
Manufacturer=Microsoft
MaxNumberControlled=0
MaxSpeed=
Name=Microsoft Kernel Debug Network Adapter
NetConnectionID=
NetConnectionStatus=
NetEnabled=
NetworkAddresses=
PermanentAddress=
PhysicalAdapter=FALSE
PNPDeviceID=ROOT\KDNIC\0000
PowerManagementCapabilities=
PowerManagementSupported=FALSE
ProductName=Microsoft Kernel Debug Network Adapter
ServiceName=kdnic
Speed=
Status=
StatusInfo=
SystemCreationClassName=Win32_ComputerSystem
SystemName=CSRF-8
TimeOfLastReset=20201011135604.500000+120


AdapterType=Ethernet 802.3
AdapterTypeId=0
AutoSense=
Availability=3
Caption=[00000001] Realtek PCIe GBE Family Controller
ConfigManagerErrorCode=0
ConfigManagerUserConfig=FALSE
CreationClassName=Win32_NetworkAdapter
Description=Realtek PCIe GBE Family Controller
DeviceID=1
ErrorCleared=
ErrorDescription=
GUID={027BE62E-5A7B-45DE-93A8-177055D03CAD}
Index=1
InstallDate=
Installed=TRUE
InterfaceIndex=2
LastErrorCode=
MACAddress=30:9C:23:E8:15:6C
Manufacturer=Realtek
MaxNumberControlled=0
MaxSpeed=
Name=Realtek PCIe GBE Family Controller
NetConnectionID=Ethernet
NetConnectionStatus=7
NetEnabled=FALSE
NetworkAddresses=
PermanentAddress=
PhysicalAdapter=TRUE
PNPDeviceID=PCI\VEN_10EC&amp;DEV_8168&amp;SUBSYS_7B241462&amp;REV_15\4&amp;574FE4C&amp;0&amp;00E0
PowerManagementCapabilities=
PowerManagementSupported=FALSE
ProductName=Realtek PCIe GBE Family Controller
ServiceName=rt640x64
Speed=9223372036854775807
Status=
StatusInfo=
SystemCreationClassName=Win32_ComputerSystem
SystemName=CSRF-8
TimeOfLastReset=20201011135604.500000+120


AdapterType=
AdapterTypeId=
AutoSense=
Availability=3
Caption=[00000002] WAN Miniport (SSTP)
ConfigManagerErrorCode=0
ConfigManagerUserConfig=FALSE
CreationClassName=Win32_NetworkAdapter
Description=WAN Miniport (SSTP)
DeviceID=2
ErrorCleared=
ErrorDescription=
GUID=
Index=2
InstallDate=
Installed=TRUE
InterfaceIndex=18
LastErrorCode=
MACAddress=
Manufacturer=Microsoft
MaxNumberControlled=0
MaxSpeed=
Name=WAN Miniport (SSTP)
NetConnectionID=
NetConnectionStatus=
NetEnabled=
NetworkAddresses=
PermanentAddress=
PhysicalAdapter=FALSE
PNPDeviceID=SWD\MSRRAS\MS_SSTPMINIPORT
PowerManagementCapabilities=
PowerManagementSupported=FALSE
ProductName=WAN Miniport (SSTP)
ServiceName=RasSstp
Speed=
Status=
StatusInfo=
SystemCreationClassName=Win32_ComputerSystem
SystemName=CSRF-8
TimeOfLastReset=20201011135604.500000+120


AdapterType=
AdapterTypeId=
AutoSense=
Availability=3
Caption=[00000003] WAN Miniport (IKEv2)
ConfigManagerErrorCode=0
ConfigManagerUserConfig=FALSE
CreationClassName=Win32_NetworkAdapter
Description=WAN Miniport (IKEv2)
DeviceID=3
ErrorCleared=
ErrorDescription=
GUID=
Index=3
InstallDate=
Installed=TRUE
InterfaceIndex=11
LastErrorCode=
MACAddress=
Manufacturer=Microsoft
MaxNumberControlled=0
MaxSpeed=
Name=WAN Miniport (IKEv2)
NetConnectionID=
NetConnectionStatus=
NetEnabled=
NetworkAddresses=
PermanentAddress=
PhysicalAdapter=FALSE
PNPDeviceID=SWD\MSRRAS\MS_AGILEVPNMINIPORT
PowerManagementCapabilities=
PowerManagementSupported=FALSE
ProductName=WAN Miniport (IKEv2)
ServiceName=RasAgileVpn
Speed=
Status=
StatusInfo=
SystemCreationClassName=Win32_ComputerSystem
SystemName=CSRF-8
TimeOfLastReset=20201011135604.500000+120


AdapterType=
AdapterTypeId=
AutoSense=
Availability=3
Caption=[00000004] WAN Miniport (L2TP)
ConfigManagerErrorCode=0
ConfigManagerUserConfig=FALSE
CreationClassName=Win32_NetworkAdapter
Description=WAN Miniport (L2TP)
DeviceID=4
ErrorCleared=
ErrorDescription=
GUID=
Index=4
InstallDate=
Installed=TRUE
InterfaceIndex=17
LastErrorCode=
MACAddress=
Manufacturer=Microsoft
MaxNumberControlled=0
MaxSpeed=
Name=WAN Miniport (L2TP)
NetConnectionID=
NetConnectionStatus=
NetEnabled=
NetworkAddresses=
PermanentAddress=
PhysicalAdapter=FALSE
PNPDeviceID=SWD\MSRRAS\MS_L2TPMINIPORT
PowerManagementCapabilities=
PowerManagementSupported=FALSE
ProductName=WAN Miniport (L2TP)
ServiceName=Rasl2tp
Speed=
Status=
StatusInfo=
SystemCreationClassName=Win32_ComputerSystem
SystemName=CSRF-8
TimeOfLastReset=20201011135604.500000+120


AdapterType=
AdapterTypeId=
AutoSense=
Availability=3
Caption=[00000005] WAN Miniport (PPTP)
ConfigManagerErrorCode=0
ConfigManagerUserConfig=FALSE
CreationClassName=Win32_NetworkAdapter
Description=WAN Miniport (PPTP)
DeviceID=5
ErrorCleared=
ErrorDescription=
GUID=
Index=5
InstallDate=
Installed=TRUE
InterfaceIndex=6
LastErrorCode=
MACAddress=
Manufacturer=Microsoft
MaxNumberControlled=0
MaxSpeed=
Name=WAN Miniport (PPTP)
NetConnectionID=
NetConnectionStatus=
NetEnabled=
NetworkAddresses=
PermanentAddress=
PhysicalAdapter=FALSE
PNPDeviceID=SWD\MSRRAS\MS_PPTPMINIPORT
PowerManagementCapabilities=
PowerManagementSupported=FALSE
ProductName=WAN Miniport (PPTP)
ServiceName=PptpMiniport
Speed=
Status=
StatusInfo=
SystemCreationClassName=Win32_ComputerSystem
SystemName=CSRF-8
TimeOfLastReset=20201011135604.500000+120


AdapterType=
AdapterTypeId=
AutoSense=
Availability=3
Caption=[00000006] WAN Miniport (PPPOE)
ConfigManagerErrorCode=0
ConfigManagerUserConfig=FALSE
CreationClassName=Win32_NetworkAdapter
Description=WAN Miniport (PPPOE)
DeviceID=6
ErrorCleared=
ErrorDescription=
GUID=
Index=6
InstallDate=
Installed=TRUE
InterfaceIndex=9
LastErrorCode=
MACAddress=
Manufacturer=Microsoft
MaxNumberControlled=0
MaxSpeed=
Name=WAN Miniport (PPPOE)
NetConnectionID=
NetConnectionStatus=
NetEnabled=
NetworkAddresses=
PermanentAddress=
PhysicalAdapter=FALSE
PNPDeviceID=SWD\MSRRAS\MS_PPPOEMINIPORT
PowerManagementCapabilities=
PowerManagementSupported=FALSE
ProductName=WAN Miniport (PPPOE)
ServiceName=RasPppoe
Speed=
Status=
StatusInfo=
SystemCreationClassName=Win32_ComputerSystem
SystemName=CSRF-8
TimeOfLastReset=20201011135604.500000+120


AdapterType=Ethernet 802.3
AdapterTypeId=0
AutoSense=
Availability=3
Caption=[00000007] WAN Miniport (IP)
ConfigManagerErrorCode=0
ConfigManagerUserConfig=FALSE
CreationClassName=Win32_NetworkAdapter
Description=WAN Miniport (IP)
DeviceID=7
ErrorCleared=
ErrorDescription=
GUID=
Index=7
InstallDate=
Installed=TRUE
InterfaceIndex=5
LastErrorCode=
MACAddress=4A:68:20:52:41:53
Manufacturer=Microsoft
MaxNumberControlled=0
MaxSpeed=
Name=WAN Miniport (IP)
NetConnectionID=
NetConnectionStatus=
NetEnabled=
NetworkAddresses=
PermanentAddress=
PhysicalAdapter=FALSE
PNPDeviceID=SWD\MSRRAS\MS_NDISWANIP
PowerManagementCapabilities=
PowerManagementSupported=FALSE
ProductName=WAN Miniport (IP)
ServiceName=NdisWan
Speed=
Status=
StatusInfo=
SystemCreationClassName=Win32_ComputerSystem
SystemName=CSRF-8
TimeOfLastReset=20201011135604.500000+120


AdapterType=Ethernet 802.3
AdapterTypeId=0
AutoSense=
Availability=3
Caption=[00000008] WAN Miniport (IPv6)
ConfigManagerErrorCode=0
ConfigManagerUserConfig=FALSE
CreationClassName=Win32_NetworkAdapter
Description=WAN Miniport (IPv6)
DeviceID=8
ErrorCleared=
ErrorDescription=
GUID=
Index=8
InstallDate=
Installed=TRUE
InterfaceIndex=16
LastErrorCode=
MACAddress=4A:68:20:52:41:53
Manufacturer=Microsoft
MaxNumberControlled=0
MaxSpeed=
Name=WAN Miniport (IPv6)
NetConnectionID=
NetConnectionStatus=
NetEnabled=
NetworkAddresses=
PermanentAddress=
PhysicalAdapter=FALSE
PNPDeviceID=SWD\MSRRAS\MS_NDISWANIPV6
PowerManagementCapabilities=
PowerManagementSupported=FALSE
ProductName=WAN Miniport (IPv6)
ServiceName=NdisWan
Speed=
Status=
StatusInfo=
SystemCreationClassName=Win32_ComputerSystem
SystemName=CSRF-8
TimeOfLastReset=20201011135604.500000+120


AdapterType=Ethernet 802.3
AdapterTypeId=0
AutoSense=
Availability=3
Caption=[00000009] WAN Miniport (Network Monitor)
ConfigManagerErrorCode=0
ConfigManagerUserConfig=FALSE
CreationClassName=Win32_NetworkAdapter
Description=WAN Miniport (Network Monitor)
DeviceID=9
ErrorCleared=
ErrorDescription=
GUID=
Index=9
InstallDate=
Installed=TRUE
InterfaceIndex=4
LastErrorCode=
MACAddress=4A:68:20:52:41:53
Manufacturer=Microsoft
MaxNumberControlled=0
MaxSpeed=
Name=WAN Miniport (Network Monitor)
NetConnectionID=
NetConnectionStatus=
NetEnabled=
NetworkAddresses=
PermanentAddress=
PhysicalAdapter=FALSE
PNPDeviceID=SWD\MSRRAS\MS_NDISWANBH
PowerManagementCapabilities=
PowerManagementSupported=FALSE
ProductName=WAN Miniport (Network Monitor)
ServiceName=NdisWan
Speed=
Status=
StatusInfo=
SystemCreationClassName=Win32_ComputerSystem
SystemName=CSRF-8
TimeOfLastReset=20201011135604.500000+120


AdapterType=
AdapterTypeId=
AutoSense=
Availability=3
Caption=[00000010] RAS Async Adapter
ConfigManagerErrorCode=
ConfigManagerUserConfig=
CreationClassName=Win32_NetworkAdapter
Description=RAS Async Adapter
DeviceID=10
ErrorCleared=
ErrorDescription=
GUID=
Index=10
InstallDate=
Installed=TRUE
InterfaceIndex=12
LastErrorCode=
MACAddress=
Manufacturer=
MaxNumberControlled=0
MaxSpeed=
Name=RAS Async Adapter
NetConnectionID=
NetConnectionStatus=
NetEnabled=
NetworkAddresses=
PermanentAddress=
PhysicalAdapter=FALSE
PNPDeviceID=
PowerManagementCapabilities=
PowerManagementSupported=FALSE
ProductName=RAS Async Adapter
ServiceName=
Speed=
Status=
StatusInfo=
SystemCreationClassName=Win32_ComputerSystem
SystemName=CSRF-8
TimeOfLastReset=20201011135604.500000+120


AdapterType=
AdapterTypeId=
AutoSense=
Availability=3
Caption=[00000011] RME MADIface USB Ethernet
ConfigManagerErrorCode=
ConfigManagerUserConfig=
CreationClassName=Win32_NetworkAdapter
Description=RME MADIface USB Ethernet
DeviceID=11
ErrorCleared=
ErrorDescription=
GUID=
Index=11
InstallDate=
Installed=TRUE
InterfaceIndex=20
LastErrorCode=
MACAddress=
Manufacturer=
MaxNumberControlled=0
MaxSpeed=
Name=RME MADIface USB Ethernet
NetConnectionID=
NetConnectionStatus=
NetEnabled=
NetworkAddresses=
PermanentAddress=
PhysicalAdapter=FALSE
PNPDeviceID=
PowerManagementCapabilities=
PowerManagementSupported=FALSE
ProductName=RME MADIface USB Ethernet
ServiceName=
Speed=
Status=
StatusInfo=
SystemCreationClassName=Win32_ComputerSystem
SystemName=CSRF-8
TimeOfLastReset=20201011135604.500000+120


AdapterType=Ethernet 802.3
AdapterTypeId=0
AutoSense=
Availability=3
Caption=[00000012] Microsoft KM-TEST Loopback Adapter
ConfigManagerErrorCode=0
ConfigManagerUserConfig=FALSE
CreationClassName=Win32_NetworkAdapter
Description=Microsoft KM-TEST Loopback Adapter
DeviceID=12
ErrorCleared=
ErrorDescription=
GUID={7B89C746-FAD2-429E-859A-1BEEA3670750}
Index=12
InstallDate=
Installed=TRUE
InterfaceIndex=13
LastErrorCode=
MACAddress=02:00:4C:4F:4F:50
Manufacturer=Microsoft
MaxNumberControlled=0
MaxSpeed=
Name=Npcap Loopback Adapter
NetConnectionID=Npcap Loopback Adapter
NetConnectionStatus=2
NetEnabled=TRUE
NetworkAddresses=
PermanentAddress=
PhysicalAdapter=TRUE
PNPDeviceID=ROOT\NET\0000
PowerManagementCapabilities=
PowerManagementSupported=FALSE
ProductName=Microsoft KM-TEST Loopback Adapter
ServiceName=kmloop
Speed=1215752192
Status=
StatusInfo=
SystemCreationClassName=Win32_ComputerSystem
SystemName=CSRF-8
TimeOfLastReset=20201011135604.500000+120


AdapterType=Ethernet 802.3
AdapterTypeId=0
AutoSense=
Availability=3
Caption=[00000014] TP-Link Wireless USB Adapter
ConfigManagerErrorCode=0
ConfigManagerUserConfig=FALSE
CreationClassName=Win32_NetworkAdapter
Description=TP-Link Wireless USB Adapter
DeviceID=14
ErrorCleared=
ErrorDescription=
GUID={F796B04D-769D-4784-8785-03A0D043B791}
Index=14
InstallDate=
Installed=TRUE
InterfaceIndex=21
LastErrorCode=
MACAddress=50:3E:AA:EF:EC:30
Manufacturer=Realtek Semiconductor Corp.
MaxNumberControlled=0
MaxSpeed=
Name=TP-Link Wireless USB Adapter
NetConnectionID=Wi-Fi
NetConnectionStatus=2
NetEnabled=TRUE
NetworkAddresses=
PermanentAddress=
PhysicalAdapter=TRUE
PNPDeviceID=USB\VID_2357&amp;PID_0109\00E04C000001
PowerManagementCapabilities=
PowerManagementSupported=FALSE
ProductName=TP-Link Wireless USB Adapter
ServiceName=RtlWlanu
Speed=144400000
Status=
StatusInfo=
SystemCreationClassName=Win32_ComputerSystem
SystemName=CSRF-8
TimeOfLastReset=20201011135604.500000+120


AdapterType=Ethernet 802.3
AdapterTypeId=0
AutoSense=
Availability=3
Caption=[00000015] Microsoft Wi-Fi Direct Virtual Adapter
ConfigManagerErrorCode=0
ConfigManagerUserConfig=FALSE
CreationClassName=Win32_NetworkAdapter
Description=Microsoft Wi-Fi Direct Virtual Adapter
DeviceID=15
ErrorCleared=
ErrorDescription=
GUID=
Index=15
InstallDate=
Installed=TRUE
InterfaceIndex=10
LastErrorCode=
MACAddress=52:3E:AA:EF:EC:30
Manufacturer=Microsoft
MaxNumberControlled=0
MaxSpeed=
Name=Microsoft Wi-Fi Direct Virtual Adapter
NetConnectionID=
NetConnectionStatus=
NetEnabled=
NetworkAddresses=
PermanentAddress=
PhysicalAdapter=FALSE
PNPDeviceID={5D624F94-8850-40C3-A3FA-A4FD2080BAF3}\VWIFIMP_WFD\6&amp;15CFD256&amp;0&amp;11
PowerManagementCapabilities=
PowerManagementSupported=FALSE
ProductName=Microsoft Wi-Fi Direct Virtual Adapter
ServiceName=vwifimp
Speed=9223372036854775807
Status=
StatusInfo=
SystemCreationClassName=Win32_ComputerSystem
SystemName=CSRF-8
TimeOfLastReset=20201011135604.500000+120


AdapterType=Ethernet 802.3
AdapterTypeId=0
AutoSense=
Availability=3
Caption=[00000016] Microsoft Wi-Fi Direct Virtual Adapter
ConfigManagerErrorCode=0
ConfigManagerUserConfig=FALSE
CreationClassName=Win32_NetworkAdapter
Description=Microsoft Wi-Fi Direct Virtual Adapter
DeviceID=16
ErrorCleared=
ErrorDescription=
GUID=
Index=16
InstallDate=
Installed=TRUE
InterfaceIndex=19
LastErrorCode=
MACAddress=50:3E:AA:EF:EC:30
Manufacturer=Microsoft
MaxNumberControlled=0
MaxSpeed=
Name=Microsoft Wi-Fi Direct Virtual Adapter #2
NetConnectionID=
NetConnectionStatus=
NetEnabled=
NetworkAddresses=
PermanentAddress=
PhysicalAdapter=FALSE
PNPDeviceID={5D624F94-8850-40C3-A3FA-A4FD2080BAF3}\VWIFIMP_WFD\6&amp;15CFD256&amp;0&amp;12
PowerManagementCapabilities=
PowerManagementSupported=FALSE
ProductName=Microsoft Wi-Fi Direct Virtual Adapter
ServiceName=vwifimp
Speed=9223372036854775807
Status=
StatusInfo=
SystemCreationClassName=Win32_ComputerSystem
SystemName=CSRF-8
TimeOfLastReset=20201011135604.500000+120


AdapterType=
AdapterTypeId=
AutoSense=
Availability=3
Caption=[00000017] TP-Link Wireless USB Adapter
ConfigManagerErrorCode=
ConfigManagerUserConfig=
CreationClassName=Win32_NetworkAdapter
Description=TP-Link Wireless USB Adapter
DeviceID=17
ErrorCleared=
ErrorDescription=
GUID=
Index=17
InstallDate=
Installed=TRUE
InterfaceIndex=14
LastErrorCode=
MACAddress=
Manufacturer=
MaxNumberControlled=0
MaxSpeed=
Name=TP-Link Wireless USB Adapter
NetConnectionID=
NetConnectionStatus=
NetEnabled=
NetworkAddresses=
PermanentAddress=
PhysicalAdapter=FALSE
PNPDeviceID=
PowerManagementCapabilities=
PowerManagementSupported=FALSE
ProductName=TP-Link Wireless USB Adapter
ServiceName=
Speed=
Status=
StatusInfo=
SystemCreationClassName=Win32_ComputerSystem
SystemName=CSRF-8
TimeOfLastReset=20201011135604.500000+120

**Comment by RichardFoss** [comment]
Sorry, just connected my Ethernet for the Ethernet adapter info:

> os.networkInterfaces()
{
  Ethernet: [
    {
      address: 'fe80::329c:23ff:fee8:156c',
      netmask: 'ffff:ffff:ffff:ffff::',
      family: 'IPv6',
      mac: '30:9c:23:e8:15:6c',
      internal: false,
      cidr: 'fe80::329c:23ff:fee8:156c/64',
      scopeid: 2
    },
    {
      address: '169.254.223.209',
      netmask: '255.255.0.0',
      family: 'IPv4',
      mac: '30:9c:23:e8:15:6c',
      internal: false,
      cidr: '169.254.223.209/16'
    }
  ],
  'Npcap Loopback Adapter': [
    {
      address: 'fe80::4cff:fe4f:4f50',
      netmask: 'ffff:ffff:ffff:ffff::',
      family: 'IPv6',
      mac: '02:00:4c:4f:4f:50',
      internal: false,
      cidr: 'fe80::4cff:fe4f:4f50/64',
      scopeid: 13
    },
    {
      address: '169.254.1.173',
      netmask: '255.255.0.0',
      family: 'IPv4',
      mac: '02:00:4c:4f:4f:50',
      internal: false,
      cidr: '169.254.1.173/16'
    }
  ],
  'Wi-Fi': [
    {
      address: '192.168.0.168',
      netmask: '255.255.255.0',
      family: 'IPv4',
      mac: '50:3e:aa:ef:ec:30',
      internal: false,
      cidr: '192.168.0.168/24'
    }

**Comment by sebhildebrandt** [comment]
Thank you @RichardFoss 

So in `si.networkInterfaces()` I return per interface:
```
    ...
    internal: true,
    virtual: false,
    type: 'wired',
    ...
```

where `type` is only `wired` and `wireless`. So basically for your use case I guess we should have here `internal` as `true` (like for a loopback 127.0.0.1 interface).

... just a first look on your data we have ...

- `node.js` (`os.interfaces()`) detects this as `internal: false`
- `wmic` tells: `PhysicalAdapter=TRUE`

So not really a clear indicator except of the interface name. Here I would extend the test if an interface is `internal`, Would this make sense?

**Comment by RichardFoss** [comment]
Thanks for looking at this Sebastian, and for your advice. Really appreciate your time!
I agree with your comment that the only clear indicator is the interface name. Not quite sure how I would extend the test, considering that the interface I want to avoid is represented as not being internal. Both the Ethernet interface I want to use and the loopback are not virtual or internal, so no way to distinguish except via the name. Looks like I just need to disregard an Interface that includes 'Loopback' in the name.

**Comment by sebhildebrandt** [comment]
@RichardFoss if I understand you correctly, marking the `npcap` interface as `internal` is not what you are thinking of?

This would be my choice, because this would be consistent to the node-js documentation of what is internal:

```
  internal <boolean> ... true if the network interface is a loopback or similar 
                         interface that is not remotely accessible; otherwise false
```

see https://nodejs.org/api/os.html#os_os_networkinterfaces

**Comment by RichardFoss** [comment]
Thanks for the feedback Sebastian, and for the link. As I see it, I could mark the npcap interface as internal, but would need to do a text search to do this. On every machine that I run my software, I would need to do the text search. So then my question would be - why not just discard the npcap loopback adapter based on the string search? What is strange to me is that it is not picked up as an internal adapter by the os.networkInterfaces() function. Maybe I'm missing something?

**Comment by sebhildebrandt** [comment]
@RichardFoss no, you are not missing something ;-) this is how node-js sees this. 

If this is OK, I would include this (detecting npcap as internal) to systeminformation. So then you only need to exclude `internal === true`... would this make sense for you?

**Comment by RichardFoss** [comment]
That would be wonderful Sebastian, thank you!!

**Comment by sebhildebrandt** [comment]
@RichardFoss I just updated the `lib/network.js` file on GitHub. Can you just update this file on jour side and have a look if this works for you?

**Comment by RichardFoss** [comment]
I'm very sorry about the slow reply Sebastian, got caught up in other work. I will try this tomorrow and let you know how it goes. Thank you very much for making the change!

**Comment by sebhildebrandt** [comment]
@RichardFoss Take your time, no stress ;-)

**Comment by RichardFoss** [comment]
Hi Sebastian, sorry to be so slow. I have made the change to network.js and am getting the following back:
Internal??:  Realtek PCIe GBE Family Controller false 
Internal??:  Npcap Loopback Adapter false
Internal??:  TP-Link Wireless USB Adapter false
Internal??:  Loopback Pseudo-Interface 1 true
So picking up on one loopback, but not the npcap?

**Comment by sebhildebrandt** [comment]
@RichardFoss Thank you ... need to check ... can you post the full output of `networkInterfaces()` ... of course you can anonymise mac addresses and IP addresses (if you need to) ...

**Comment by RichardFoss** [comment]
How would that be different from what I gave previously?

**Comment by sebhildebrandt** [comment]
@RichardFoss I mean the full output of systeminformations `si.netwoekInterfaces()` (not from node.js). I do not see this in your previous posts

**Comment by RichardFoss** [comment]
Sorry, its your call to networkInterfaces - will do that!

**Comment by RichardFoss** [comment]
0: {iface: "Ethernet", ifaceName: "Realtek PCIe GBE Family Controller", ip4: "169.254.223.209", ip4subnet: "255.255.0.0", ip6: "fe80::329c:23ff:fee8:156c", …}
1: {iface: "Npcap Loopback Adapter", ifaceName: "Npcap Loopback Adapter", ip4: "169.254.1.173", ip4subnet: "255.255.0.0", ip6: "fe80::4cff:fe4f:4f50", …}
2: {iface: "Wi-Fi", ifaceName: "TP-Link Wireless USB Adapter", ip4: "192.168.0.168", ip4subnet: "255.255.255.0", ip6: "", …}
3: {iface: "Loopback Pseudo-Interface 1", ifaceName: "Loopback Pseudo-Interface 1", ip4: "127.0.0.1", ip4subnet: "255.0.0.0", ip6: "::1", …}

**Comment by RichardFoss** [comment]
Shall I go into the objects?

**Comment by sebhildebrandt** [comment]
@RichardFoss Thank you ... that is fine for me ... will come back later.

**Comment by RichardFoss** [comment]
Great, thanks for your help!

**Comment by sebhildebrandt** [comment]
@RichardFoss my fault ... can you again download latest version of `/lib/network.js`and test one more time?

**Comment by RichardFoss** [comment]
Sure, will do Sebastian!

**Comment by RichardFoss** [comment]
It works!! Thanks!! Really need to buy you coffee now :)  
Internal??:  Realtek PCIe GBE Family Controller false
indexjs.js:478 Internal??:  Npcap Loopback Adapter true
indexjs.js:478 Internal??:  TP-Link Wireless USB Adapter false
indexjs.js:478 Internal??:  Loopback Pseudo-Interface 1 true

**Comment by sebhildebrandt** [comment]
@RichardFoss that's good to hear, that this works now. Just published a new npm version 4.27.9
Closing it for now.

**Comment by RichardFoss** [comment]
Wonderful, thanks Sebastian!
